From 900e22d8e024f4113243ea16cba2cfea1a41cba0 Mon Sep 17 00:00:00 2001
From: Vamsi Krishna Brahmajosyula <vamsi-krishna.brahmajosyula@broadcom.com>
Date: Sun, 5 Apr 2026 18:21:27 +0000
Subject: [PATCH 17/19] perf print events: Fix compiler issues with newer gilbc

In newer glibc, strstr function returns const char * based
on input constant string, handle this by casting to char *

Signed-off-by: Vamsi Krishna Brahmajosyula <vamsi-krishna.brahmajosyula@broadcom.com>
---
 tools/perf/util/print-events.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/tools/perf/util/print-events.c b/tools/perf/util/print-events.c
index 3624a3c79..850c7acfc 100644
--- a/tools/perf/util/print-events.c
+++ b/tools/perf/util/print-events.c
@@ -387,7 +387,7 @@ void print_symbol_events(const struct print_callbacks *print_cb, void *print_sta
 	}
 
 	strlist__for_each_entry(nd, evt_name_list) {
-		char *alias = strstr(nd->s, " OR ");
+		char *alias = strstr((char *)nd->s, " OR ");
 
 		if (alias) {
 			*alias = '\0';
-- 
2.43.7

